### ISSUE TITLE
Suggestion: use Services[Characteristic] to make code more pythonic.

### ISSUE THREAD

**Issue by schinckel** [ikalchev/HAP-python#40]
This could then raise a `KeyError`, rather than an `AssertionError` when a non-existent characteristic is accessed.

**Comment by schinckel** [comment]
Indeed, I thought I remembered that assertions are removed when optimising: https://stackoverflow.com/a/1838411/188

**Comment by ikalchev** [comment]
This is something new for me, I didn't know asserts can get ignored. I will give it some though.
